src/components/Input/Input.tsx

```diff
@@ -1,54 +1,60 @@
-import * as PropTypes from 'prop-types'
 import * as React from 'react'
+import * as PropTypes from 'prop-types'
+import * as cx from 'classnames'
 import * as _ from 'lodash'
 
 import {
   AutoControlledComponent,
-  createHTMLInput,
   customPropTypes,
-  getUnhandledProps,
+  IRenderResultConfig,
   partitionHTMLProps,
 } from '../../lib'
-import Icon from '../Icon'
-import { ComponentVariablesInput, ComponentPartStyle } from '../../../types/theme'
 import {
-  ComponentEventHandler,
   Extendable,
-  ReactChildren,
-  ShorthandRenderFunction,
   ShorthandValue,
+  ShorthandRenderFunction,
+  ComponentEventHandler,
 } from '../../../types/utils'
+import { ComponentPartStyle, ComponentVariablesInput } from '../../../types/theme'
+import Icon from '../Icon'
+import Slot from '../Slot'
+import Ref from '../Ref'
 
 export interface IInputProps {
   as?: any
-  children?: ReactChildren
   className?: string
   clearable?: boolean
-  defaultValue?: string | number
+  defaultValue?: React.ReactText
   fluid?: boolean
   icon?: ShorthandValue
   inline?: boolean
   input?: ShorthandValue
   onChange?: ComponentEventHandler<IInputProps>
-  value?: string | number
-  type?: string
   renderIcon?: ShorthandRenderFunction
   renderInput?: ShorthandRenderFunction
-  styles?: ComponentPartStyle
+  renderWrapper?: ShorthandRenderFunction
+  styles?: ComponentPartStyle<IInputProps, any>
+  type?: string
+  value?: React.ReactText
   variables?: ComponentVariablesInput
+  wrapper?: ShorthandValue
+}
+
+export interface IInputState {
+  value?: React.ReactText
 }
 
 /**
  * An Input
  * @accessibility
  * For good screen reader experience set aria-label or aria-labelledby attribute for input.
  *
- *
  * Other considerations:
  *  - if input is search, then use "role='search'"
- *
  */
-class Input extends AutoControlledComponent<Extendable<IInputProps>, any> {
+class Input extends AutoControlledComponent<Extendable<IInputProps>, IInputState> {
+  private inputRef: HTMLInputElement
```

Comment:
this is not ref, name like `inputElement` or, better, `inputDomElement` would be more descriptive here